ocaml/xapi/message_forwarding.ml

```diff
@@ -2030,6 +2030,31 @@ functor
             forward_vm_op ~local_fn ~__context ~vm ~remote_fn
         )
 
+      let call_host_plugin ~__context ~vm ~plugin ~fn ~args =
+        info
+          "VM.call_host_plugin: VM = '%s'; plugin = '%s'; fn = '%s'; args = [ \
+           'hidden' ]"
+          (vm_uuid ~__context vm) plugin fn ;
+        let local_fn = Local.VM.call_host_plugin ~vm ~plugin ~fn ~args in
+        let remote_fn = Client.VM.call_host_plugin ~vm ~plugin ~fn ~args in
+        let power_state = Db.VM.get_power_state ~__context ~self:vm in
+        if power_state <> `Running then
```

Comment:
Is there any reason that the VM can't be paused?

Reply:
I the explanation should be added as a comment, then

Reply:
If we want to operate on some VM that is not running, we can also use Host.call_plugin. In that case the VM has no associoted host and it does not make sense to use the VM to find a host, A paused VM is some middle ground - it has a host but it it not reactive. If this is a useful case, we could some flag in the future. 

I all a comment.